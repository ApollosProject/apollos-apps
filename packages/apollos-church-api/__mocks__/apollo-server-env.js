import fetch from 'jest-fetch-mock';

import { Constants } from 'apollos-church-api/src/connectors/rock';
import * as apolloDatasourceMocks from './apollo-datasource-mocks';
import * as rockMocks from './rock-api-mocks';

const apolloServerEnv = require.requireActual('apollo-server-env');

// eslint-disable-next-line
const Response = apolloServerEnv.Response;

const resolveWith = (data, url) =>
  Promise.resolve(
    new Response(JSON.stringify(data), {
      url,
      status: 200,
      statusText: 'OK',
      headers: new apolloServerEnv.Headers({
        'Content-Type': 'application/json',
      }),
    })
  );

fetch.mockLiveDataSourceApis = () => {
  fetch.mockImplementation((request) => {
    if (request.url.match('/api/v1/events/current')) {
      return resolveWith(apolloDatasourceMocks.liveStreamLive(), request.url);
    }
    if (request.url.match('/passage/html')) {
      return resolveWith(apolloDatasourceMocks.ESVScripture(), request.url);
    }
    return Promise.reject();
  });
};

fetch.mockRockDataSourceAPI = () => {
  fetch.mockImplementation((request) => {
    let { url } = request;
    url = decodeURI(url);
    if (!url.match(Constants.ROCK_API)) return Promise.reject();

    if (url.match('api/ContentChannels/\\d')) {
      return resolveWith(rockMocks.contentChannel());
    }

    if (url.match('api/ContentChannels')) {
      return resolveWith([rockMocks.contentChannel()]);
    }

    if (url.match('api/ContentChannelItems/test-case-no-attributes')) {
      const mock = rockMocks.contentItem();
      mock.AttributeValues = {};
      mock.Attributes = {};
      return resolveWith(mock);
    }

    if (url.match('api/ContentChannelItems/\\d')) {
      return resolveWith(rockMocks.contentItem());
    }

    if (url.match('api/ContentChannelItems')) {
      return resolveWith([rockMocks.contentItem()]);
    }

    if (url.match('api/ContentChannelItemAssociations')) {
      return resolveWith([rockMocks.contentChannelItemAssociation()]);
    }

    if (url.match('api/Auth/Login')) {
      const body = JSON.parse(request.body);
      const response = new Response('');
      if (body.Password === 'good') {
        response.headers.set('set-cookie', 'some cookie');
        return Promise.resolve(response);
      }
      return Promise.reject({ ...response, status: 401 });
    }

    if (url.match('api/People/GetCurrentPerson')) {
      return resolveWith(rockMocks.people());
    }

    if (url.match('api/People/\\d')) {
      return resolveWith(rockMocks.people());
    }

    if (url.match('api/People')) {
      if (request.method === 'POST') {
        const { Email } = JSON.parse(request.body);
        if (!Email) {
          const response = new Response('');
          response.status = 400;
          return Promise.reject(response);
        }
        return resolveWith({ personId: 35 });
      }

      return resolveWith([rockMocks.people()]);
    }

    if (url.match('api/UserLogins')) {
      if (request.method === 'POST') {
        const { UserName } = JSON.parse(request.body);
        if (!UserName) {
          const response = new Response('');
          response.status = 400;
          return Promise.reject(response);
        }
        return resolveWith({ id: 21 });
      }
      const identity = url // identity = UserName
        .split('eq')
        .pop()
        .trim(' '); // EXAMPLE URL: /api/UserLogins?$filter=UserName eq 'isaac.hardy@newspring.cc'

      if (!identity) {
        const response = new Response('');
        response.status = 400;
        return Promise.reject(response);
      }
      if (identity === `'isaac.hardy@newspring.cc'`)
        return resolveWith([rockMocks.userLogins()]);

      return resolveWith([]);
    }

    return Promise.reject();
  });
};

const apolloServerEnvMocked = {
  ...apolloServerEnv,
  Response: fetch.Response,
  fetch,
};

module.exports = apolloServerEnvMocked;
