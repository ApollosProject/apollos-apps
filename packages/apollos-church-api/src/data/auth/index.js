import { gql } from 'apollo-server';
import { createGlobalId } from '../node';

// export { default as model } from './model';
export { default as dataSource } from './data-source';

export const schema = gql`
  type AuthenticatedUser {
    id: ID!
    profile: Person
  }

  type Authentication {
    user: AuthenticatedUser
    token: String
  }

  extend type Mutation {
    authenticate(identity: String!, password: String!): Authentication
    registerPerson(email: String!, password: String!): Authentication
  }

  extend type Query {
    currentUser: AuthenticatedUser
  }
`;

export const resolver = {
  Query: {
    currentUser: (root, args, { dataSources }) =>
      dataSources.Auth.getCurrentPerson(),
  },
  AuthenticatedUser: {
    id: ({ id }, args, context, { parentType }) =>
      createGlobalId(id, parentType.name),
    profile: (authUser) => authUser,
  },
  Authentication: {
    user: (root, args, { dataSources }) => dataSources.Auth.getCurrentPerson(),
  },
  Mutation: {
    authenticate: (root, { identity, password }, { dataSources }) =>
      dataSources.Auth.authenticate({ identity, password }),
    registerPerson: (root, args, { dataSources }) =>
      dataSources.Auth.registerPerson(args),
  },
};
