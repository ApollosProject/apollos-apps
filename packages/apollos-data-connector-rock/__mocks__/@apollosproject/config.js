const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK_MAPPINGS: {
    CONTENT_ITEM: {
      ContentSeriesContentItem: {
        EntityType: 'ContentChannelItem',
      },
      DevotionalContentItem: {
        EntityType: 'SomeOtherEntity',
      },
      MediaContentItem: {
        EntityType: 'ContentChannelItem',
      },
      UniversalContentItem: {
        EntityType: 'ContentChannelItem',
      },
      ContentItem: {
        EntityType: 'ContentChannelItem',
      },
    },
  },
  ROCK: {
    TIMEZONE: 'America/New_York',
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
  },
});

export default ApolloServer;
