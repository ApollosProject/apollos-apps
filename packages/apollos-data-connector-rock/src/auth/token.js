import { AuthenticationError } from 'apollo-server';
import jwt from 'jsonwebtoken';

export const secret = process.env.SECRET || 'ASea$2gadj#asd0';

export const parseToken = (token) => jwt.verify(token, secret);

export const registerToken = (token) => {
  try {
    const { cookie, sessionId } = parseToken(token);

    return {
      userToken: token,
      rockCookie: cookie,
      sessionId,
    };
  } catch (e) {
    if (e instanceof jwt.TokenExpiredError) {
      return {};
    }
    throw new AuthenticationError('Invalid token');
  }
};

export const generateToken = (params) =>
  jwt.sign({ ...params }, secret, { expiresIn: '400d' });
