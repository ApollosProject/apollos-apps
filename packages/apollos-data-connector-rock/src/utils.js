import { AuthenticationError } from 'apollo-server';

export const enforceProtocol = (uri) =>
  uri.startsWith('//') ? `https:${uri}` : uri;

export const createImageUrlFromGuid = (uri, Config) =>
  uri.split('-').length === 5
    ? `${
        Config.ROCK.IMAGE_URL || `${Config.ROCK.URL}/GetImage.ashx`
      }?guid=${uri}`
    : enforceProtocol(uri);

export const fieldsAsObject = (fields) =>
  fields.reduce(
    (accum, { field, value }) => ({
      ...accum,
      [field]: typeof value === 'string' ? value.trim() : value,
    }),
    {}
  );

export const enforceCurrentUser = (func) => async (
  root,
  args,
  context,
  info
) => {
  try {
    const currentPerson = await context.dataSources.Person.getCurrentPerson();
    if (root.id !== currentPerson.originId) {
      return null;
    }
  } catch (e) {
    if (!(e instanceof AuthenticationError)) {
      throw e;
    }
    return null;
  }
  return func(root, args, context, info);
};
