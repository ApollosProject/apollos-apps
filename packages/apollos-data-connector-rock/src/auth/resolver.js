import { createGlobalId } from '@apollosproject/server-core';

const resolver = {
  Query: {
    currentUser: (root, args, { dataSources }) =>
      dataSources.Auth.getCurrentPerson(),
  },
  AuthenticatedUser: {
    id: ({ id }, args, context, { parentType }) =>
      createGlobalId(id, parentType.name),
    profile: (authUser) => authUser,
    rock: (rockDetails) => rockDetails,
    rockToken: (root, args, context) => context.rockCookie,
  },
  RockPersonDetails: {
    authToken: (root, args, { dataSources }) => dataSources.Auth.getAuthToken(),
    authCookie: (root, args, { rockCookie }) => rockCookie,
  },
  Authentication: {
    user: (root, args, { dataSources }) => dataSources.Auth.getCurrentPerson(),
  },
  Mutation: {
    authenticate: (root, { identity, password }, { dataSources }) =>
      dataSources.Auth.authenticate({ identity, password }),
    changePassword: (root, { password }, { dataSources }) =>
      dataSources.Auth.changePassword({ password }),
    registerPerson: (root, args, { dataSources }) =>
      dataSources.Auth.registerPerson(args),
  },
};

export default resolver;
