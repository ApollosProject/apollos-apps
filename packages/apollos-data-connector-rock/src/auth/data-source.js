import { AuthenticationError } from 'apollo-server';
import { fetch, Request } from 'apollo-server-env';
import moment from 'moment';
import RockApolloDataSource from '@apollosproject/rock-apollo-data-source';

import { generateToken, registerToken } from './token';

export default class AuthDataSource extends RockApolloDataSource {
  resource = 'Auth';

  rockCookie = null;

  userToken = null;

  initialize(config) {
    super.initialize(config);
    if (config.context.rockCookie) {
      // fetches the current person
      // this method will try to cache the current person on the context
      // removing the need to fetch each and every time
      this.getCurrentPerson();
    }
  }

  getCurrentPerson = async ({ cookie = null } = { cookie: null }) => {
    const { rockCookie, currentPerson } = this.context;
    const userCookie = cookie || rockCookie;

    if (currentPerson) {
      return currentPerson;
    }

    if (userCookie) {
      try {
        const request = await this.request('People/GetCurrentPerson').get({
          options: {
            headers: { cookie: userCookie, 'Authorization-Token': null },
          },
        });
        this.context.currentPerson = request;
        return request;
      } catch (e) {
        throw new AuthenticationError('Invalid user cookie');
      }
    }
    throw new AuthenticationError('Must be logged in');
  };

  fetchUserCookie = async (Username, Password) => {
    try {
      // We use `new Response` rather than string/options b/c if conforms more closely with ApolloRESTDataSource
      // (makes mocking in tests WAY easier to use `new Request` as an input in both places)
      const response = await fetch(
        new Request(`${this.baseURL}/Auth/Login`, {
          method: 'POST',
          body: JSON.stringify({
            Username,
            Password,
          }),
          headers: {
            'Content-Type': 'Application/Json',
          },
        })
      );
      if (response.status >= 400) throw new AuthenticationError();
      const cookie = response.headers.get('set-cookie');
      return cookie;
    } catch (err) {
      throw new AuthenticationError('Invalid Credentials');
    }
  };

  createSession = async ({ cookie }) => {
    const currentUser = await this.getCurrentPerson({ cookie });
    return this.post('/InteractionSessions', {
      PersonAliasId: currentUser.primaryAliasId,
    });
  };

  authenticate = async ({ identity, password }) => {
    try {
      const cookie = await this.fetchUserCookie(identity, password);
      const sessionId = await this.createSession({ cookie });
      const token = generateToken({ cookie, sessionId });
      const { userToken, rockCookie } = registerToken(token);
      this.context.rockCookie = rockCookie;
      this.context.userToken = userToken;
      this.context.sessionId = sessionId;
      return { token, rockCookie };
    } catch (e) {
      throw e;
    }
  };

  personExists = async ({ identity }) => {
    const hasUserName = await this.request(
      `/UserLogins?$filter=UserName eq '${identity}'`
    ).get();

    if (hasUserName.length) {
      return true;
    }
    return false;
  };

  createUserProfile = async (props = {}) => {
    try {
      const { email } = props;

      return await this.post('/People', {
        Email: email,
        IsSystem: false, // Required by Rock
        Gender: 0, // Required by Rock
      });
    } catch (err) {
      throw new Error('Unable to create profile!');
    }
  };

  createUserLogin = async (props = {}) => {
    try {
      const { email, password, personId } = props;

      return await this.post('/UserLogins', {
        PersonId: personId,
        EntityTypeId: 27, // A default setting we use in Rock-person-creation-flow
        UserName: email,
        PlainTextPassword: password,
        LastLoginDateTime: `${moment().toISOString()}`,
      });
    } catch (err) {
      throw new Error('Unable to create user login!');
    }
  };

  changePassword = async ({ password }) => {
    const currentUser = await this.getCurrentPerson();
    const { email, id } = currentUser;
    const login = await this.request('/UserLogins')
      .filter(`UserName eq '${email}'`)
      .first();

    if (login) {
      await this.delete(`/UserLogins/${login.id}`);
    }
    await this.createUserLogin({
      personId: id,
      email,
      password,
    });
    return this.authenticate({
      identity: email,
      password,
    });
  };

  registerPerson = async ({ email, password }) => {
    const personExists = await this.personExists({ identity: email });
    if (personExists) throw new Error('User already exists!');

    const personId = await this.createUserProfile({ email });
    await this.createUserLogin({
      email,
      password,
      personId,
    });

    const token = await this.authenticate({ identity: email, password });
    return token;
  };

  getAuthToken = async () => {
    const { rockCookie } = this.context;
    if (!rockCookie) throw new AuthenticationError('Must be logged in');

    // TODO remove this safety check and less secure implementation
    // once core Rock has the GetCurrentPersonImpersonationToken endpoint
    try {
      const token = await this.request(
        `People/GetCurrentPersonImpersonationToken?expireDateTime=${moment()
          .add(1, 'weeks')
          .toISOString()}`
      ).get({
        options: {
          headers: { cookie: rockCookie, 'Authorization-Token': null },
        },
      });
      return token;
    } catch (e) {
      console.warn(
        'Using deprecated Rock endpoint, upgrade Rock to v10 when available.'
      );
      const { id } = await this.getCurrentPerson();
      const param = await this.request(
        `People/GetImpersonationParameter?personId=${id}&expireDateTime=${moment()
          .add(1, 'weeks')
          .toISOString()}`
      ).get();
      return param.split('=')[1];
    }
  };
}
