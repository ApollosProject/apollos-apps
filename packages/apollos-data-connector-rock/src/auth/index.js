import { createGlobalId } from '@apollosproject/server-core';
import { get } from 'lodash';
import { registerToken } from './token';

export { registerToken, generateToken } from './token';
export { authSchema as schema } from '@apollosproject/data-schema';
export { default as dataSource } from './data-source';

export const contextMiddleware = ({ req, context: ctx }) => {
  if (get(req, 'headers.authorization')) {
    const { userToken, rockCookie, sessionId } = registerToken(
      req.headers.authorization
    );
    if (sessionId) {
      return {
        ...ctx,
        userToken,
        rockCookie,
        sessionId,
      };
    }
  }
  return ctx;
};

export const resolver = {
  Query: {
    currentUser: (root, args, { dataSources }) =>
      dataSources.Auth.getCurrentPerson(),
  },
  AuthenticatedUser: {
    id: ({ id }, args, context, { parentType }) =>
      createGlobalId(id, parentType.name),
    profile: (authUser) => authUser,
  },
  Authentication: {
    user: (root, args, { dataSources }) => dataSources.Auth.getCurrentPerson(),
  },
  Mutation: {
    authenticate: (root, { identity, password }, { dataSources }) =>
      dataSources.Auth.authenticate({ identity, password }),
    changePassword: (root, { password }, { dataSources }) =>
      dataSources.Auth.changePassword({ password }),
    registerPerson: (root, args, { dataSources }) =>
      dataSources.Auth.registerPerson(args),
  },
};
