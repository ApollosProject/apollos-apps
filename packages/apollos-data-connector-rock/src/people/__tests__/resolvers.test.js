import { graphql } from 'graphql';
import { fetch } from 'apollo-server-env';
import { AuthenticationError } from 'apollo-server';
import ApollosConfig from '@apollosproject/config';
import { createGlobalId } from '@apollosproject/server-core';
import { createTestHelpers } from '@apollosproject/server-core/lib/testUtils';
import {
  peopleSchema,
  mediaSchema,
  authSchema,
  deviceSchema,
} from '@apollosproject/data-schema';
import { generateToken, registerToken } from '../../auth';
// we import the root-level schema and resolver so we test the entire integration:
import * as Person from '../index';
import { PersonalDevice } from '../../index';
import authMock from '../../authMock';
import { enforceCurrentUser } from '../../utils';

const Auth = { schema: authSchema, dataSource: authMock };
const { getContext, getSchema } = createTestHelpers({
  Person,
  Auth,
  PersonalDevice,
});

ApollosConfig.loadJs({
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
  },
});

describe('Person', () => {
  let schema;
  let context;
  beforeEach(() => {
    fetch.resetMocks();
    fetch.mockRockDataSourceAPI();
    schema = getSchema([peopleSchema, mediaSchema, deviceSchema]);
    context = getContext();
  });

  it("updates a user's attributes, if there is a current user", async () => {
    const query = `
      mutation {
        updateProfileField(input: { field: FirstName, value: "Richard" }) {
          firstName
          id
        }
      }
    `;
    const { userToken, rockCookie } = registerToken(
      generateToken({ cookie: 'some-cookie' })
    );
    context.userToken = userToken;
    context.rockCookie = rockCookie;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('updates multiple fields', async () => {
    const query = `
      mutation {
        updateProfileFields(input: [
          { field: FirstName, value: "Richard" },
          { field: LastName, value: "Walkerton" }
        ]) {
          firstName
          lastName
          id
        }
      }
    `;
    const { userToken, rockCookie } = registerToken(
      generateToken({ cookie: 'some-cookie' })
    );
    context.userToken = userToken;
    context.rockCookie = rockCookie;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it("updates a user's gender", async () => {
    const query = `
      mutation {
        updateProfileFields(input: [
          { field: Gender, value: "Male" },
        ]) {
          id
          gender
        }
      }
    `;
    const { userToken, rockCookie } = registerToken(
      generateToken({ cookie: 'some-cookie' })
    );
    context.userToken = userToken;
    context.rockCookie = rockCookie;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it("fails to update a user's attributes, without a current user", async () => {
    const query = `
      mutation {
        updateProfileField(input: { field: FirstName, value: "Richard" }) {
          firstName
          id
        }
      }
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets a single person when querying by root node', async () => {
    const query = `
      query {
        node(
          id: "${createGlobalId(51, 'Person')}"
        ) {
          ... on Person {
            id
            firstName
            lastName
            nickName
            email
            photo {
              uri
            }
            devices {
              id
              pushId
              notificationsEnabled
            }
          }
        }
      }
    `;

    context.dataSources.PersonalDevice.request = jest.fn(() => ({
      filter: jest.fn(() => ({
        get: jest.fn(() =>
          Promise.resolve([
            {
              id: '1',
              deviceRegistrationId: 'abc-123',
              notificationsEnabled: true,
            },
          ])
        ),
      })),
    }));

    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });
});

describe('enforceCurrentUser', () => {
  it('will return a field if the queried user is the current user', async () => {
    const context = {
      dataSources: { Auth: { getCurrentPerson: () => ({ id: 1 }) } },
    };
    const resultFunc = enforceCurrentUser(({ firstName }) => firstName);

    const result = await resultFunc(
      { firstName: 'John', id: 1 },
      {},
      context,
      {}
    );
    expect(result).toEqual('John');
  });

  it("won't return a field with no user", async () => {
    const context = {
      dataSources: {
        Auth: { getCurrentPerson: () => throw new AuthenticationError() },
      },
    };
    const resultFunc = enforceCurrentUser(({ firstName }) => firstName);

    const result = await resultFunc(
      { firstName: 'John', id: 1 },
      {},
      context,
      {}
    );
    expect(result).toEqual(null);
  });

  it("won't return a field with a different user", async () => {
    const context = {
      dataSources: { Auth: { getCurrentPerson: () => ({ id: 9 }) } },
    };
    const resultFunc = enforceCurrentUser(({ firstName }) => firstName);

    const result = await resultFunc(
      { firstName: 'John', id: 1 },
      {},
      context,
      {}
    );
    expect(result).toEqual(null);
  });

  it("won't swallow non-auth errors", async () => {
    const context = {
      dataSources: { Auth: { getCurrentPerson: () => ({ id: 9 }) } },
    };
    const resultFunc = enforceCurrentUser(({ firstName }) => firstName);

    const result = await resultFunc(
      { firstName: 'John', id: 1 },
      {},
      context,
      {}
    );
    expect(result).toEqual(null);
  });

  it("won't swallow non-auth errors", () => {
    const context = {
      dataSources: {
        Auth: { getCurrentPerson: () => throw new Error('Random Error') },
      },
    };
    const resultFunc = enforceCurrentUser(({ firstName }) => firstName);

    const result = resultFunc({ firstName: 'John', id: 1 }, {}, context, {});
    expect(result).rejects.toThrowErrorMatchingSnapshot();
  });
});
