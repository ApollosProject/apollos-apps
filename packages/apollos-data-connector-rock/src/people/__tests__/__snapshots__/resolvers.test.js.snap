// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person fails to update a user's attributes, without a current user 1`] = `
Object {
  "data": Object {
    "updateProfileField": Object {
      "firstName": "Richard",
      "id": "Person:057b0758543270d1cb21784cac4d3f5c",
    },
  },
}
`;

exports[`Person gets a single person when querying by root node 1`] = `
Object {
  "data": Object {
    "node": Object {
      "devices": Array [
        Object {
          "id": "Device:559b23fd0aa90e81b1c023e72e230fa1",
          "notificationsEnabled": true,
          "pushId": "abc-123",
        },
      ],
      "email": "isaac.hardy@newspring.cc",
      "firstName": "Isaac",
      "id": "Person:057b0758543270d1cb21784cac4d3f5c",
      "lastName": "Hardy",
      "nickName": "Isaac",
      "photo": Object {
        "uri": "https://apollosrock.newspring.cc:443/GetImage.ashx?guid=60fd5f35-3167-4c26-9a30-d44937287b87",
      },
    },
  },
}
`;

exports[`Person updates a user's attributes, if there is a current user 1`] = `
Object {
  "data": Object {
    "updateProfileField": Object {
      "firstName": "Richard",
      "id": "Person:057b0758543270d1cb21784cac4d3f5c",
    },
  },
}
`;

exports[`Person updates a user's gender 1`] = `
Object {
  "data": Object {
    "updateProfileFields": Object {
      "gender": "Male",
      "id": "Person:057b0758543270d1cb21784cac4d3f5c",
    },
  },
}
`;

exports[`Person updates multiple fields 1`] = `
Object {
  "data": Object {
    "updateProfileFields": Object {
      "firstName": "Richard",
      "id": "Person:057b0758543270d1cb21784cac4d3f5c",
      "lastName": "Walkerton",
    },
  },
}
`;

exports[`enforceCurrentUser won't swallow non-auth errors 1`] = `"Random Error"`;
